collect_app/src/main/java/org/odk/collect/android/widgets/utilities/ActivityGeoDataRequester.java

```diff
@@ -0,0 +1,108 @@
+package org.odk.collect.android.widgets.utilities;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+
+import org.javarosa.form.api.FormEntryPrompt;
+import org.odk.collect.android.activities.GeoPointActivity;
+import org.odk.collect.android.activities.GeoPointMapActivity;
+import org.odk.collect.android.activities.GeoPolyActivity;
+import org.odk.collect.android.listeners.PermissionListener;
+import org.odk.collect.android.utilities.ApplicationConstants;
+import org.odk.collect.android.utilities.PermissionUtils;
+import org.odk.collect.android.utilities.WidgetAppearanceUtils;
+
+public class ActivityGeoDataRequester {
```

Comment:
This should still implement the interface so that the widget depends on an abstraction rather than an implementation.

Reply:
I misinterpreted the last comment. The interface would be good here.